### PR TITLE
Initialize the new section for Java GUI developmment

### PR DESCRIPTION
**Pull request by jdneo**
@nickzhums Please let me know if you want to add more content in the new section for Java GUI development.

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

**Comment by nickzhums** [comment]
what would be the title of the section? I'm thinking about something like "GUI Applications (JavaFX, Swing)"

File: docs/java/java-gui.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Working with GUI application in Visual Studio Code
+
+You can develop Java GUI applications in Visual Studio Code easily. To achieve that, you need to install the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), which includes all the required extensions to develop Java GUI application.
+
+If you run into any issues when using the features below, you can contact us by entering an [issue](https://github.com/microsoft/vscode-java-pack/issues).
+
+## Develop JavaFX Application
+
+Now you can simply create a new JavaFX application will very few steps in Visual Studio Code. Open the command palette **Command Palette**  (`kb(workbench.action.showCommands)`) and then select the command: **Java: Create Java Project**. You will find the option **JavaFX** in the list, which will help you scaffold a new JavaFX project via Maven archetype.
```
Comment (nickzhums) [inline docs/java/java-gui.md:9]:
```suggestion
Now you can simply create a new JavaFX application with just a few steps in Visual Studio Code. Open the command palette **Command Palette**  (`kb(workbench.action.showCommands)`) and then select the command: **Java: Create Java Project**. You will find the option **JavaFX** in the list, which will help you scaffold a new JavaFX project via Maven archetype.
```

File: docs/java/java-gui.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Working with GUI application in Visual Studio Code
+
+You can develop Java GUI applications in Visual Studio Code easily. To achieve that, you need to install the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), which includes all the required extensions to develop Java GUI application.
+
+If you run into any issues when using the features below, you can contact us by entering an [issue](https://github.com/microsoft/vscode-java-pack/issues).
```
Comment (nickzhums) [inline docs/java/java-gui.md:5]: can we also provide some guidance for unmanaged projects? e.g. what are the steps they will need to take

File: docs/java/java-gui.md (hunk 1 above)
Comment (nickzhums) [reply docs/java/java-gui.md:9]: also suggest to break this content down into several steps:
Step 1: Install extension pack 
Step 2: xx
Step 3: yy

File: docs/java/java-gui.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Develop AWT Application
+
+By default, the types from AWT are hidden. You may notice that the code completion is not working when you are developing an AWT application. To enable that, you can open the command palette **Command Palette**  (`kb(workbench.action.showCommands)`) and then select the command: **Java: Help Center**. Go to the **Student** section and click `Enable AWT Development`.
+
+> Note: This action will update a setting in workspace level, so please make sure a workspace in opened in your Visual Studio Code.
+
+## Develop Swing Application
+
+Swing application development is supported by default. You can directly write your Swing application code without any setup steps.
```
Comment (nickzhums) [inline docs/java/java-gui.md:27]: can we provide link to some sample projects for Swing/awt (can be on our github repos?)

Comment (jdneo) [reply docs/java/java-gui.md:27]: What about these two links:

AWT: https://docs.oracle.com/javase/8/docs/technotes/guides/awt/index.html
Swing: https://docs.oracle.com/javase/tutorial/uiswing/examples/components/index.html

Comment (nickzhums) [reply docs/java/java-gui.md:27]: let's try to provide a sample project if possible (preferably can be directly opened in VS Code)

**Comment by jdneo** [comment]
Hi @gregvanl, this is a new section we want to add into the Java part. I noticed that each page will have a set of metadata at the beginning. But I have no idea about how to get the things like `ContentId`. Will that be added by someone else?

**Comment by gregvanl** [comment]
Hi @jdneo, you don't need to worry about the metadata section, I can fill it in when I do an edit pass after we merge your PR.

One question is where in the table of contents would you like the new topic. The Order: value sets the order in the Java TOC. 
Maybe down under Application Servers?

image

**Comment by jdneo** [comment]
@gregvanl Thank you for the quick answer.

> Maybe down under Application Servers?

I'll sync with @nickzhums and fill the required metadata then. Under Application Servers looks good to me.

**Comment by nickzhums** [comment]
below Application Server and above Java on Azure :)

**Comment by nickzhums** [comment]
discussed with @jdneo, we want to put it below Java on Azure and above "Extensions", thanks !

**Comment by jdneo** [comment]
@nickzhums Comments addressed. The PR will keep as draft until the new extension pack for java get released.

**Comment by nickzhums** [comment]
looks good to me - thanks!

**Review by nickzhums — APPROVED** [review]

File: docs/java/java-gui.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  public static void main(String args[]){
+    new AwtExample();
+  }
+}
+```
+
+## Develop Swing Application
+
+Swing application development is supported by default. You can directly write your Swing application code without any setup steps.
```
Comment (nickzhums) [inline docs/java/java-gui.md:82]:
```suggestion
Swing application development is supported by default. You can directly write your Swing application code without any setup.
```

File: docs/java/java-gui.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+To run the JavaFX application, you can open the `Maven` explorer, expand `hellofx` > `Plugins` > `javafx` and run the Maven goal: `javafx:run`.
+
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-gui/run-javafx.mp4" type="video/mp4">
+</video>
+
+### More JavaFX examples
+
+There are more JavaFX project examples, which cover Gradle and unmanaged folder projects in Visual Studio Code as well. You can find them [here](https://github.com/openjfx/samples/tree/master/IDE/VSCode). Each sample has documentation about how to run it as well.
```
Comment (nickzhums) [inline docs/java/java-gui.md:44]:
```suggestion
More JavaFX project examples can be found at [here](https://github.com/openjfx/samples/tree/master/IDE/VSCode), which covers different project structures (such as Gradle and unmanaged projects). There is documentation on each sample to describe how to run the program.
```

**Comment by jdneo** [comment]
Mark the PR as `ready` since the java pack has been released yesterday.

**Pull request merged**